File: com-mini-util/src/main/java/com/mini/util/digest/SHA256Util.java
/**
 * Created the com.cfinal.util.digest.SHA256Util.java
 * @created 2017年5月16日 下午4:31:08
 * @version 1.0.0
 */
package com.mini.util.digest;

import java.io.InputStream;

/**
 * com.cfinal.util.digest.SHA256Util.java
 *
 * @author XChao
 */
public class SHA256Util {
    private static final DigestUtil instance = new DigestUtil("SHA-256");


    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(String data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(String data) {
        return instance.encode(data);
    }

    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(byte[] data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(byte[] data) {
        return instance.encode(data);
    }

    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static byte[] encodeToByte(InputStream data) {
        return instance.encodeToByte(data);
    }

    /**
     * SHA256加密
     *
     * @param data 加密数据
     * @return 加密结果
     */
    public static String encode(InputStream data) {
        return instance.encode(data);
    }

}

File: sn.mini.kotlin.util/src/main/java/sn/mini/kotlin/util/degest/SHA256Util.kt
/**
 * Created the com.cfinal.util.digest.SHA256Util.java
 *
 * @created 2017年5月16日 下午4:31:08
 * @version 1.0.0
 */
package sn.mini.kotlin.util.degest

import java.io.InputStream

/**
 * com.cfinal.util.digest.SHA256Util.java
 *
 * @author XChao
 */
object SHA256Util {
    private val instance = DigestUtil("SHA-256")


    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encodeToByte(data: String): ByteArray = instance.encodeToByte(data)

    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encode(data: String): String = instance.encode(data)

    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encodeToByte(data: ByteArray): ByteArray = instance.encodeToByte(data)

    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encode(data: ByteArray): String = instance.encode(data)

    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encodeToByte(data: InputStream): ByteArray = instance.encodeToByte(data)

    /**
     * SHA256加密
     *
     * @param data
     * @return
     */
    fun encode(data: InputStream): String = instance.encode(data)

}
